backend/tests/test_course_build_processor.py

```diff
@@ -4,7 +4,8 @@
 import json
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-
```

Comment:
I didn't look too deeply, but I think the following two tests are currently failing sometimes due to non-determinism somewhere:

* `test_course_build_status_completed` in `test_course_build_processor.py`
* `test_get_course_build_by_academic_year` in `test_endpoints_moodle.py`

It might be worth a look to see if we can figure out / address for our future selves. Perhaps that can be split out into a different issue, though, as neither are directly related to K12-27 / this PR :man_shrugging: .

Reply:
:thinking: -- It seems like an anti-pattern to change application logic to get assertions to work properly for `test_get_course_build_by_academic_year`. Perhaps we can take the approach we do in other repos where instead of making assumptions on order we just confirm length and inclusion. 

For `test_get_course_build_by_academic_year`, perhaps we should remove the ordering bit altogether and just query the fixture row that has status `completed`. Alternatively, we could have the fixture return the test data as dict and use that in places to remove the indirect "lookups" in the DB :man_shrugging: .

Reply:
Resolving this comment as the items will be resolved in [K12-69](https://openstax.atlassian.net/browse/K12-69).